### PR TITLE
payment enabled

### PR DESCRIPTION
**Pull request by tjw0612**
## Why

This pull request is needed because:

payment is possible

### CODE REVIEW

File: app/controllers/payments_controller.rb [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+      if rest.promo_status == 1
+        total_sum += (Food.find(ord["dishId"].to_i).price_cents * 0.8).round * ord["quantity"].to_i
+      else
+        total_sum += Food.find(ord["dishId"].to_i).price_cents * ord["quantity"].to_i
+      end
+    end
+
     # teddy = Teddy.find(params[:teddy_id])
     # order  = Order.create!(teddy: teddy, teddy_sku: teddy.sku, amount: teddy.price, state: 'pending', user: current_user)
+    # params["total_price"]
```
Comment (mxng) [inline app/controllers/payments_controller.rb]: delete here if unneeded

File: app/views/food_orders/_create_order.html.erb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <%= form_tag payments_path do %>
   <%# <%= hidden_field_tag 'food_order_id', @food_order.id %>
   <div class="d-flex justify-content-center mb-3">
+    <%# <%= hidden_field_tag 'total_price', @money %>
```
Comment (mxng) [inline app/views/food_orders/_create_order.html.erb]: delete?

**Review by mxng — CHANGES_REQUESTED** [review]
Need to delete the commented out lines

File: app/controllers/payments_controller.rb [hunk 3]
```diff
@@ -1,18 +1,37 @@
...
-        name: "testing",
-        amount: "500",
+        name: "#{Order.last.details[0]['restaurant']} - #{current_user.email}",
+        amount: total_sum,
         currency: 'sgd',
         quantity: 1
       }],
       mode: 'payment',
-      success_url: orders_url(root_path),
+      success_url: "http://localhost:3000/success",
```
Comment (TeeWhyKay) [inline app/controllers/payments_controller.rb]: this will not work in production

**Review by mxng — APPROVED** [review]
Looks ok

**Review by TeeWhyKay — APPROVED** [review]
well done, thanks for the hard work

**Pull request merged**